.github/workflows/nightlies.yml

```diff
@@ -0,0 +1,57 @@
+name: Upload nightly wheels to Anaconda Cloud
+
+on:
+  # Run daily at 1:23 UTC to upload nightly wheels to Anaconda Cloud
+  schedule:
+  - cron: '23 1 * * *'
+  # Run on demand with workflow dispatch
+  workflow_dispatch:
+
+jobs:
+  upload_nightly_wheels:
+    name: Upload nightly wheels to Anaconda Cloud
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.x'
+
+      # c.f. https://github.com/actions/download-artifact/issues/3#issuecomment-1017141067
+      - name: Download wheel artifacts from last build on 'main'
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          PROJECT_REPO=matplotlib/matplotlib
+          BRANCH=main
+          WORKFLOW_NAME="cibuildwheel.yml"
+          ARTIFACT_NAME=wheels
+
+          gh run --repo "${PROJECT_REPO}" list --branch "${BRANCH}" --workflow "${WORKFLOW_NAME}" --json status,databaseId > runs.json
+          # Get latest completed build of wheels
+          RUN_ID=$(cat runs.json | jq -c '[ .[] | select(.status == "completed") ] | sort_by(.databaseId) | reverse | .[0].databaseId')
+          gh run --repo "${PROJECT_REPO}" download "${RUN_ID}" --name "${ARTIFACT_NAME}"
+
+          mkdir dist
+          mv *.whl dist/
+          ls -l dist/
+
+      - name: Install anaconda-client
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          # c.f. https://github.com/Anaconda-Platform/anaconda-client/issues/540
+          python -m pip install git+https://github.com/Anaconda-Server/anaconda-client
+          python -m pip list
+
+      - name: List files to upload
+        run: |
+          echo "# x86_64.manylinux wheels to upload to Anaconda Cloud:"
+          find dist -iname *x86_64.manylinux*.whl
+
+      - name: Upload x86_64.manylinux wheels to Anaconda Cloud as nightlies
+        run: |
+          anaconda --token ${{ secrets.ANACONDA_ORG_UPLOAD_TOKEN }} upload \
```

Comment:
c.f. https://github.com/matplotlib/matplotlib/pull/21637#issuecomment-971947384

Reply:
`ANACONDA_ORG_UPLOAD_TOKEN` is still the right secret but I do not think we will know if this works until we merge it.

Reply:
Not sure if it is worth it, but you could manually download a wheel and then upload it with the token to test in advance that the `ANACONDA_ORG_UPLOAD_TOKEN` is valid. Though given that this PR enables workflow dispatch if there was something wrong with the token you could just change the value of the repo secret and rerun the job (which runs in under 1 minute) so probably not worth the time to test it out before.